komand/server.py

```diff
@@ -177,6 +177,91 @@ def plugin_info():
+            ---
+            get:
+              summary: Get list of plugin actions
+              description: Get InsightConnect plugin all actions
+              responses:
+                200:
+                  description: InsightConnect Plugin actions list to be returned
+                  schema:
+                    type: array
+            """
+            action_list = []
+            for action in self.plugin.actions.keys():
+                action_list.append(action)
+            return json.dumps(action_list, indent=2)
+
+        @app.route("/triggers")
+        def triggers():
+            """Plugin triggers list endpoint.
+            ---
+            get:
+              summary: Get list of plugin triggers
+              description: Get InsightConnect plugin all triggers
+              responses:
+                200:
+                  description: InsightConnect Plugin triggers list to be returned
+                  schema:
+                    type: array
+            """
+            trigger_list = []
+            for action in self.plugin.triggers.keys():
+                trigger_list.append(action)
+            return json.dumps(trigger_list, indent=2)
+
+        @app.route("/status")
+        def status():
+            """Web service status endpoint
+            ---
+            get:
+              summary: Get web service status
+              description: Get web service status
+              responses:
+                200:
+                  description: Status to be returned
+                  schema:
+                    type: object
+            """
+            # TODO: Add logic to figure out status (Ready, Running, Down) of web service.
+            return jsonify({"status": "Ready"})
+
+        @app.route("/spec")
+        def plugin_spec():
+            """Plugin spec details endpoint.
+            ---
+            get:
+              summary: Get plugin spec details
+              description: Get plugin specification
+              parameters:
+                - in: query
+                  name: format
+                  type: string
+                  description: Format to return plugin spec; defaults to JSON
+                  enum: [json, yaml]
+              responses:
+                200:
+                  description: Plugin specification to be returned
+                  schema:
+                    type: object
+                404:
+                  description: The specified format is not supported
+            """
+            with open("/python/src/plugin.spec.yaml", 'r') as p_spec:
+                plugin_spec = p_spec.read()
+
+            format_ = request.args.get('format', 'json')
+
+            if format_ == "json":
+                return jsonify(yaml.safe_load(plugin_spec))
+            elif format_ == "yaml":
+                return plugin_spec
+            else:
+                return make_response(jsonify({'error': 'The specified format is not supported'}), 404)
```

Comment:
```suggestion
                return make_response(jsonify({"error": "The specified format is not supported"}), 400)
```

400 since the query parameter is bad, not the path itself.

Reply:
This is a good catch and an oversight by me on another PR.  What about a `422` for `Unprocessable Entity` since the parameter sent may very well be format/content properly but just wrong value?